test/lit/passes/type-generalizing.wast

```diff
@@ -0,0 +1,219 @@
+;; NOTE: Assertions have been generated by update_lit_checks.py --all-items and should not be edited.
+;; RUN: foreach %s %t wasm-opt --dce --experimental-type-generalizing -all -S -o - | filecheck %s
```

Comment:
I would be happy to scan unreachable code, but CFGWalker currently does not produce unreachable basic blocks, so there is currently nothing in the CFG to scan. Do you think it's worth adding support to CFGWalker to produce unreachable basic blocks?

Reply:
Ah, yeah, we avoid creating unreachable basic blocks (at least obvious ones). It's not worth changing that I think. However, if this pass only runs properly after DCE, we'd need to run DCE internally inside it to avoid problems.

Reply:
Hmm, that means that any pass that uses the framework will have to internally run DCE first, which seems unfortunate. I'll run DCE inside this pass for now, but maybe it will be worth having opt-in unreachable blocks in the future.